### PR TITLE
solution

### PR DESCRIPTION
**Pull request by andrii-koptiev**
[DEMO LINK](https://andrii-koptiev.github.io/react_clock/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+type State = {
+  isClockVisible: boolean;
+  clockName: number;
+};
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+class App extends React.Component<{}, State> {
+  state = {
```
Comment (JHesher) [inline src/App.tsx:11]:
```suggestion
  state: State = {
```

File: src/components/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  name: number;
+};
+
+type State = {
+  date: Date;
+  timerId: NodeJS.Timer | undefined;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
```
Comment (JHesher) [inline src/components/Clock.tsx:13]:
```suggestion
  state: State = {
```

**Review by JHesher — APPROVED** [review]
looks great🔥

File: src/App.tsx (hunk 1 above)
Comment (andrii-koptiev) [reply src/App.tsx]: Ty. Done.

File: src/components/Clock.tsx (hunk 2 above)
Comment (andrii-koptiev) [reply src/components/Clock.tsx]: Ty. Done.

File: src/components/Clock.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    // eslint-disable-next-line no-console
+    console.log(this.state.date.toLocaleTimeString());
+  };
+
+  render() {
+    return (
+      <>
+        {this.state.date.toLocaleTimeString()}
+      </>
+    );
```
Comment (Elena-brv) [inline src/components/Clock.tsx:46]:
```suggestion
    return this.state.date.toLocaleTimeString();
    );
```

**Review by Elena-brv — APPROVED** [review]

**Pull request closed without merging**